### PR TITLE
Added Comments to Stock Pages

### PR DESCRIPTION
**Pull request by javidangarcia**
- Added comments database model and endpoints for creating and getting comments.
- Added comments component that allows users to write a comment and send on each stock page.

https://github.com/javidangarcia/stock-zone/assets/89328228/29d30e00-e619-4b28-98f8-043c2576bd1a

https://github.com/javidangarcia/stock-zone/assets/89328228/55b2800a-3541-4ea6-a03f-1dcb44dcdca1

Screenshot 2023-07-18 at 9 27 45 AM

### CODE REVIEW

File: backend/models/comment.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    content: {
+        type: DataTypes.STRING(2000),
+        allowNull: false
+    }
+});
+
+Comment.belongsTo(User);
+Comment.belongsTo(Stock);
+User.hasMany(Comment);
+Stock.hasMany(Comment);
```
Comment (vandanasub) [inline backend/models/comment.js:21]: data model looks good to me

File: frontend/src/components/StockData/StockData.jsx [hunk 2]
```diff
@@ -144,6 +145,7 @@ export default function StockData() {
                         </div>
                         <p>{stockData.description}</p>
                         <p>Sector: {stockData.sector}</p>
+                        <Comments ticker={ticker} />
```
Comment (vandanasub) [inline frontend/src/components/StockData/StockData.jsx:148]: Good, I like how the Comments section is its own component

File: backend/routes/comments.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import express from "express";
+import { Comment } from "../models/comment.js";
+import { Stock } from "../models/stock.js";
+import { User } from "../models/user.js";
+
+const router = express.Router();
+
+router.get("/comments/:ticker", async (req, res) => {
```
Comment (vandanasub) [inline backend/routes/comments.js:8]: Curious, any reason to do /comments/:ticker instead of the other way like :ticker/comments?

**Review by vandanasub — COMMENTED** [review]
Nice, the comments component on the UI looks awesome!

File: backend/routes/comments.js (hunk 3 above)
Comment (javidangarcia) [reply backend/routes/comments.js:8]: It looks better to me the first way but the other way makes sense.

**Comment by BriptimusPrimus** [comment]
It would be good to add a video where user A sees user's B comments on Stock X. Can you add that to the test plan?

File: frontend/src/components/Comments/Comments.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            };
+
+            const response = await axios.post(
+                "http://localhost:3000/comments",
+                commentData,
+                { withCredentials: true, validateStatus: () => true }
+            );
+
+            if (response.status === 200) {
+                fetchComments();
```
Comment (BriptimusPrimus) [inline frontend/src/components/Comments/Comments.jsx:55]: This additional network call can be avoided with an "optimistic update". We have a successful response, so we assume the comment went in to the DB, thus, we can add it to the current state.

**Pull request merged**